### PR TITLE
fix: Dasherize admin sales APIs and fix admin/sales/location API

### LINKED ISSUE #5527: Dasherize admin sales apis and fix admin/sales/location api.
**Describe the bug**

- Most of the Admin sales api are not dasherized.
- admin/sales/location returns the data without id.

These fixes are required by fossasia/open-event-frontend/issues/1807

### PR DESCRIPTION
**Pull request by YogeshSharma0201**
Fixes #5527
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
- Admin sales apis are dasherized.
- location_name is used as id, earlier it was not returning any id, which lead to errors on frontend. (Since location_name is unique i have used that as id, please do suggest any better alternative approch).
 ```
locations = self.session.query(
            Event.location_name,
            Event.location_name.label('id')) \
                .group_by(Event.location_name) \
                .filter(Event.location_name != None) \
                .cte()
```

### CODE REVIEW

File: app/api/admin_sales/locations.py [hunk 1]
```diff
@@ -63,9 +65,12 @@ class AdminSalesByLocationList(ResourceList):
...
 
     def query(self, _):
-        locations = self.session.query(Event.location_name) \
-                                .group_by(Event.location_name) \
-                                .cte()
+        locations = self.session.query(
+            Event.location_name,
+            Event.location_name.label('id')) \
+                .group_by(Event.location_name) \
+                .filter(Event.location_name != None) \
```
[the remark this answers is not in the document]
Comment (CosmicCoder96) [reply app/api/admin_sales/locations.py]: Please replace this with
```python
.filter(Event.location_name is not None) \
```

**Comment by iamareebjamal** [comment]
Then use `query.filter(User.name.isnot(None))`

**Comment by YogeshSharma0201** [comment]
@iamareebjamal i have changed that.

**Comment by iamareebjamal** [comment]
Please change the PR title to be more descriptive and according to the guidelines

**Pull request merged**